lib/Provider/PrivacyIDEAProvider.php

```diff
@@ -68,61 +68,62 @@ public function __construct(IAppConfig $appConfig, LoggerInterface $logger, IReq
 	 */
 	public function getTemplate(IUser $user): Template
 	{
-		if ($this->isTwoFactorAuthEnabledForUser($user) === false) {
+		if (!$this->isTwoFactorAuthEnabledForUser($user)) {
 			$this->session->set('piNoAuthRequired', true);
 			$this->verifyChallenge($user, '');
 		} else {
 			$this->session->set('piAllowCreatingPIInstance', true);
 			$this->pi = $this->createPrivacyIDEAInstance();
-		}
-		$authenticationFlow = $this->getAppValue('piSelectedAuthFlow', 'piAuthFlowDefault');
-		$this->log('debug', 'Selected authentication flow: ' . $authenticationFlow);
-		$username = $user->getUID();
-		$headers = [];
-		$headersFromConfig = $this->getAppValue('piForwardHeaders', '');
-		if (!empty($headersFromConfig)) {
-			$headers = $this->getHeadersToForward($headersFromConfig);
-		}
 
-		if ($authenticationFlow === 'piAuthFlowTriggerChallenge') {
-			if (!empty($this->pi)) {
-				if (!$this->pi->serviceAccountAvailable()) {
-					$this->log('error', 'Service account name or password is not set in config. Cannot trigger the challenges.');
-				} else {
-					if ($this->session->get('piTriggerChallengeDone') !== true) {
-						try {
-							$response = $this->pi->triggerChallenge($username, $headers);
-							$this->session->set('piTriggerChallengeDone', true);
-							if ($response !== null) {
-								$this->processPIResponse($response);
-							} else {
-								$this->log('error', 'No response from privacyIDEA server for triggerchallenge.');
+			$authenticationFlow = $this->getAppValue('piSelectedAuthFlow', 'piAuthFlowDefault');
+			$this->log('debug', 'Selected authentication flow: ' . $authenticationFlow);
+			$username = $user->getUID();
+			$headers = [];
+			$headersFromConfig = $this->getAppValue('piForwardHeaders', '');
+			if (!empty($headersFromConfig)) {
+				$headers = $this->getHeadersToForward($headersFromConfig);
+			}
+
+			if ($authenticationFlow === 'piAuthFlowTriggerChallenge') {
+				if (!empty($this->pi)) {
```

Comment:
this is too much nesting of ifs. refactor to have the checks for `$this->pi` before evaluating the authentication flow to exit early.